### PR TITLE
chore(attachments) Cleanup dangling event attachment files

### PR DESCRIPTION
**Pull request by markstory**
The previous deletion code didn't cascade into EventAttachment files because event attachments were deleted with a bulk SQL query. This has left many unreferenced files in production that we would like to purge.

This migration looks at files older than the oldest event attachment and removes files if they are no longer referenced by an attachment. This is slower but safer.

### CODE REVIEW

File: src/sentry/migrations/0093_cleanup_unreferenced_event_files.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        "unreal.logs",
+    ]
+    file_query = (File.objects
+        .filter(timestamp__lt=oldest_attachment['date_added__min'])
+        .filter(type__in=attachment_types))
+
+    for f in RangeQuerySetWrapperWithProgressBar(file_query):
+        # Double check that the file is not referenced.
+        if not EventAttachment.objects.filter(file=f).exists():
+            f.delete()
```
Comment (wedamija) [inline src/sentry/migrations/0093_cleanup_unreferenced_event_files.py:38]: Do we also need to delete the related `FileBlob` rows here? And if so, do we expect that this should also delete the file itself from storage?

Comment (JTCunning) [reply src/sentry/migrations/0093_cleanup_unreferenced_event_files.py:38]: Short answer, no.
FileBlobs are a many-to-many relationship with FileBlobIndex and File.
The cleanup to remove dangling Fileblobs is done [here](https://github.com/getsentry/sentry/blob/master/src/sentry/runner/commands/cleanup.py#L328-L352).

**Comment by JTCunning** [comment]
Looks logically good to me. I'll let someone else apply the final +1.

File: src/sentry/migrations/0093_cleanup_unreferenced_event_files.py (hunk 1 above)
Comment (markstory) [reply src/sentry/migrations/0093_cleanup_unreferenced_event_files.py:38]: The File model is a curious beast. It uses relationship cascades to delete the blob models [which overload delete() to clean up storage](https://github.com/getsentry/sentry/blob/master/src/sentry/models/file.py#L263-L286).

Similar style deletions for [releases use the .delete() method on ReleaseFile](https://github.com/getsentry/sentry/blob/e0c247c17c59a72f2241d972933dfd3203ecf0be/src/sentry/deletions/defaults/organization.py#L46) and rely on django cascading deletes into File and FileBlob which is what I used as prior art for this.

Comment (wedamija) [reply src/sentry/migrations/0093_cleanup_unreferenced_event_files.py:38]: I don't think we can rely on the custom `delete` from the `FileBlob` in migrations, these models are just 'fake' versions of the models. So I don't think that will actually happen. It also would not cascade across a M2M relationship, and afaik the FK relationship is deprecated.

Anyway, based on what @JTCunning said the file blobs will be cleaned up via a separate process, so we should be good here.

**Review by wedamija — APPROVED** [review]
LGTM, there's an index on `timestamp` so it should be efficient to use that as a filter.

File: src/sentry/migrations/0093_cleanup_unreferenced_event_files.py (hunk 1 above)
Comment (markstory) [reply src/sentry/migrations/0093_cleanup_unreferenced_event_files.py:38]: I had forgotten that django uses fake models in migrations, the blob cleanup will catch the 2nd tier of orphans :smile:

**Pull request merged**